[2024-11-22T18:42:12.267+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-22T18:42:12.354+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T18:40:00+00:00 [queued]>
[2024-11-22T18:42:12.424+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T18:40:00+00:00 [queued]>
[2024-11-22T18:42:12.428+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-22T18:42:12.496+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-22 18:40:00+00:00
[2024-11-22T18:42:12.526+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=215) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-22T18:42:12.530+0000] {standard_task_runner.py:72} INFO - Started process 219 to run task
[2024-11-22T18:42:12.531+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-22T18:40:00+00:00', '--job-id', '368', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmph0ommn3a']
[2024-11-22T18:42:12.542+0000] {standard_task_runner.py:105} INFO - Job 368: Subtask process_transaction
[2024-11-22T18:42:12.812+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T18:40:00+00:00 [running]> on host c6af8dbaf1e5
[2024-11-22T18:42:13.267+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-22T18:40:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-22T18:40:00+00:00'
[2024-11-22T18:42:13.271+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-22T18:42:13.341+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-22T18:42:13.389+0000] {fraud_detection_dag.py:77} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[280923],"data":[[2264937662466770,"fraud_Monahan, Bogisich and Ledner","misc_pos",1.62,"Juan","Sherman","M","5939 Garcia Forges Suite 297","San Antonio","TX",78214,29.3641,-98.4924,1595797,"Land","1995-10-17","091567e542d9142c26c26199f9cbfe41",30.272806,-98.579808,0,1732300925463]]}
[2024-11-22T18:42:13.393+0000] {fraud_detection_dag.py:85} INFO - Données JSON parsées avec succès
[2024-11-22T18:42:13.406+0000] {fraud_detection_dag.py:89} INFO - DataFrame créé avec 1 lignes
[2024-11-22T18:42:13.414+0000] {fraud_detection_dag.py:93} INFO - Colonnes renommées avec succès
[2024-11-22T18:42:13.416+0000] {fraud_detection_dag.py:94} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-22T18:42:13.421+0000] {fraud_detection_dag.py:98} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-22T18:42:13.537+0000] {fraud_detection_dag.py:107} INFO - Transformation des données...
[2024-11-22T18:42:15.664+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-22T18:42:15.732+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-22T18:42:15.742+0000] {fraud_detection_dag.py:119} INFO - Prediction de fraude: NORMAL, Probabilité: 32.40%
[2024-11-22T18:42:15.864+0000] {fraud_detection_dag.py:125} INFO - Prédiction: NORMAL (Probabilité: 32.40%)
[2024-11-22T18:42:15.869+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-22T18:42:15.870+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-22T18:42:15.872+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-22T18:42:15.931+0000] {skipmixin.py:278} INFO - Skipping tasks [('skip_processing', -1), ('notify_fraud', -1)]
[2024-11-22T18:42:16.081+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-22T18:42:16.083+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-22T18:40:00+00:00, execution_date=20241122T184000, start_date=20241122T184212, end_date=20241122T184216
[2024-11-22T18:42:16.195+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-22T18:42:16.354+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-22T18:42:16.394+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
